### PR TITLE
calendar#521 Add modify event

### PR DESCRIPTION
**Pull request by rezk2ll**

### CODE REVIEW

File: lib/api/calendars.ts [hunk 1]
```diff
@@ -42,3 +42,11 @@ export const getInbox = (client: DAVClient) => async (userId: string): Promise<C
     ),
   }));
 }
+
+export const modifyEvent = (client: DAVClient) => async (eventPath: string, event: CalendarEventObject): Promise<string> => {
+  return await client.requestText({
+    url: `${BASE_PATH}/${eventPath}`,
```
Comment (huy-ta) [inline lib/api/calendars.ts:48]: You should use `url-join`. Btw, I forgot to do the same for the `getInbox` method. Could you please do it for me 😛.

File: test/api/calendar.test.ts [hunk 2]
```diff
@@ -67,3 +68,59 @@ describe('the getInbox method', () => {
...
+CALSCALE:GREGORIAN\r\n\
+BEGIN:VEVENT\r\n\
+UID:123456\r\n\
+SUMMARY:test\r\n\
+LOCATION:some location\r\n\
+DESCRIPTION:simple description\r\n\
+DTSTART:20210315T100000\r\n\
+DTEND:20210315T110000\r\n\
+END:VEVENT\r\n\
+END:VCALENDAR',
```
Comment (huy-ta) [inline test/api/calendar.test.ts:120]: You're also testing the integrity of the `dav-parser` here?

Comment (huy-ta) [reply test/api/calendar.test.ts:120]: So this is more like an integration test. Is this what you intended to do?

Comment (rezk2ll) [reply test/api/calendar.test.ts:120]: nope, the translator is tested in the `dav-parser` library with a variation of calendar event objects.
I'm here simply testing the parameters that the `requestText` was called with.

Comment (huy-ta) [reply test/api/calendar.test.ts:120]: Then you should mock the `translate` function of `dav-parser` here.

Comment (huy-ta) [reply test/api/calendar.test.ts:120]: The true unit test requires the mocking of `url-join` also :joy:.

Comment (huy-ta) [reply test/api/calendar.test.ts:120]: But hold on did you already do the same for the tests for `getInbox`? If so... proceed :joy:.

Comment (rezk2ll) [reply test/api/calendar.test.ts:120]: no need to do anything related to `url-join` in the tests, it works out of the box

**Comment by asmaNode** [comment]
package-lock.json file should'nt be committed

**Review by asmaNode — APPROVED** [review]

**Comment by rezk2ll** [comment]
> package-lock.json file should'nt be committed

needed for the CI, i will change how the CI installs the packages to not use the file

**Review by huy-ta — APPROVED** [review]

**Pull request merged**